react/features/subtitles/components/AbstractClosedCaptionButton.js

```diff
@@ -74,15 +74,26 @@ export class AbstractClosedCaptionButton
  * {@code AbstractClosedCaptionButton} component.
  *
  * @param {Object} state - The redux state.
+ * @param {Object} ownProps - The properties explicitly passed to the component
+ * instance.
  * @private
  * @returns {{
- *     __requestingSubtitles: boolean
+ *     _requestingSubtitles: boolean,
+ *     visible: boolean
  * }}
  */
-export function _abstractMapStateToProps(state: Object) {
+export function _abstractMapStateToProps(state: Object, ownProps: Object) {
     const { _requestingSubtitles } = state['features/subtitles'];
+    const { transcribingEnabled } = state['features/base/config'];
+
+    let { visible } = ownProps;
+
+    if (typeof visible === 'undefined') {
+        visible = transcribingEnabled;
+    }
```

Comment:
If I understand the following correctly, it's trying to default `visible` to `transcribingEnabled`.
````
    let { visible } = ownProps;

     if (typeof visible === 'undefined') {
         visible = transcribingEnabled;
     }
````

Isn't the intention more obviously expressed by object destructuring with default values?
````
const { visible = transcribingEnabled } = ownProps;
````

Which can even be further optimized as follows because the default value is computed only if needed:
````
const { visible = state['features/base/config'].transcribingEnabled } = ownProps;
````

Reply:
It does optimise it indeed, but lack of the 'undefined' check makes it less obvious what is the goal here. Maybe it's just me. I would probably put it with an extra comment that it's about a default value.

Reply:
My reasoning comes from the fact that Zoli for example took extra care in similar case to explain the situation with long comment:
https://github.com/jitsi/jitsi-meet/blob/master/react/features/recording/components/Recording/AbstractRecordButton.js#L116

Btw. the last optimisation does not fit into one line and doesn't look great when split, so I'll go with:
`const { visible = transcribingEnabled } = ownProps;`

Reply:
We don't need to explain ES6+ language features, a draft or standard rigorously specifies and documents their semantics.

Reply:
It just takes much more time to process such complex one line statement, but it's subjective. Do you want me to remove the comment ?

Reply:
I remember it was taking me so much time reading such neatly optimised code and trying to decode what each of the parts are doing exactly. It was the code around here (it has change a bit since then):
https://github.com/jitsi/jitsi-meet/blob/master/react/features/base/lib-jitsi-meet/native/RTCPeerConnection.js#L285
It's still people reading that code and not interpreters.